slint-sc: specify inheritance semantics and expand the property definition (#12710)
slint-ui/slint: docs/astro/src/content/docs/reference/language/file-structure.mdx

The identifier following the `component` keyword is the *name* of the component. \{#sls.file.component.name}

The first form defines a component with no explicit base;
the second form defines a component that *inherits* from the element type named by `Base`. \{#sls.file.component.inherits}

A component that inherits extends its base:
an instance of the component is an element of type `Base`. \{#sls.file.component.inherits.meaning}

The [bindings](../bindings/) of the component body apply to that element,

> and the elements instantiated in the component body are its children. \{#sls.file.component.inherits.body-applies}

Reviewer: `its` is really ambiguous here. Here's an attempt at clarifying.

```suggestion
and the elements instantiated in the component body become children of the element. \{#sls.file.component.inherits.body-applies}
```
